test/ruby/test_regexp.rb

```diff
@@ -723,6 +723,12 @@ def test_initialize
     assert_raise(RegexpError) { Regexp.new("((?<v>))\\g<0>") }
   end
 
+  def test_initialize_from_regex_memory_corruption
+    10_000.times do
+      Regexp.new(Regexp.new("(?<name>)"))
+    end
```

Comment:
@nobu does this test look ok? I am not sure if there is an existing pattern for tests like this.

Reply:
One possible way is `assert_ruby_status`, to assert it doesn't crash.
```ruby
    assert_ruby_status([], "#{<<-"begin;"}\n#{<<-'end;'}")
    begin;
      10_000.times do
        Regexp.new(Regexp.new("(?<name>)"))
      end
    end;
```